fake_modules/qwik-city/middleware/vercel-edge/index.mjs

```diff
@@ -62,18 +77,49 @@ function createQwikCity(opts) {
       const notFoundHtml = getNotFound(url.pathname);
       return new Response(notFoundHtml, {
         status: 404,
-        headers: { "Content-Type": "text/html; charset=utf-8", "X-Not-Found": url.pathname }
+        headers: {
+          "Content-Type": "text/html; charset=utf-8",
+          "X-Not-Found": url.pathname,
+        },
       });
     } catch (e) {
       console.error(e);
       return new Response(String(e || "Error"), {
         status: 500,
-        headers: { "Content-Type": "text/plain; charset=utf-8", "X-Error": "vercel-edge" }
+        headers: {
+          "Content-Type": "text/plain; charset=utf-8",
+          "X-Error": "vercel-edge",
+        },
+      });
+    }
+  }
+  async function onVercelServerlessRequest(req, res) {
+    if (req.url) {
+      const [_path, search] = req.url.split("?");
+      const params = new URLSearchParams(search);
+      let pathname = params.get("__pathname");
+
+      if (pathname) {
+        params.delete("__pathname");
+        pathname = pathname.replace(/\/+/g, "/");
+        req.url = `${pathname}?${params}`;
+      }
+    }
+    let request;
+    try {
+      request = await getRequest({
+        base: `https://${req.headers.host}`,
+        request: req,
       });
+    } catch (err) {
+      res.statusCode = err.status || 400;
+      return res.end("Invalid request body");
     }
+    const response = await onVercelEdgeRequest(request);
+
+    setResponse(res, response);
   }
-  return onVercelEdgeRequest;
+
+  return env === "edge" ? onVercelEdgeRequest : onVercelServerlessRequest;
```

Comment:
based on the `env` config provided by user in `createQwikCity` middleware, return the relevant handler for the env